### ISSUE TITLE
Video mode hopping in 15 Khz mode

### ISSUE THREAD

**Issue by egrath** [somhi/PCXT_DeMiSTify#10]
**Issue:** 
Video mode hops when in 15 Khz mode, exactly the same way it did on the MENU core before it got fixed

https://github.com/mist-devel/mist-binaries/issues/140

**Solution:**
It was solved by aligning vsync to hsync, maybe this would be possible here too.

**Comment by somhi** [comment]
Thanks for reporting.    Unfortunately, I don't have any upscaler to reproduce this behaviour.  Without upscaler I don't appreciate any hoping.  So it's very difficult for me to debug this, apart from that I don't have too much experience in this field.   
Let's see if someone else could help.

**Comment by egrath** [comment]
OK, will try to find the part in the source where the sync signals are generated and build a modified version with the changes from the previously mentioned ticket incorporated. Will keep this ticket updated about the progress.

**Comment by somhi** [comment]
Have a look at Gyurco comments [here](https://atari-forum.com/viewtopic.php?p=446485#p446485)

**Comment by somhi** [comment]
Ok, just for the sake of testing, I've created three 15khz branches on the repository.  You can test them all.
I have enabled for MiST a DEBUG menu on OSD where you can test all combinations and see if you see some change.
on the -2 and -3, OSD is not displaying well, but as these are just random tests it doesn't matter.

-3 on some video modes, the VS is shown in middle of screen.    I haven't seen though any bending lines at the upper part of the screen. That is a problem I have in my 15 kHz setup.in some games.  I don't really use 15 kHz in a daily use.

Let me know your results @egrath , and I will delete all these testing branches.

**Comment by somhi** [comment]
let me know your results, and I will delete all these testing branches.

**Comment by egrath** [comment]
I've built cores from all three branches and tested them - it made no difference. Results:

**PCXT_mist_230421.rbf**:
- RetroTink 5X fails to pick up the signal correctly, image swirls around. Made a video of it: https://photos.app.goo.gl/ckiA25tY1CHhMzWJA
- RetroTink's signal status switches rapidly between 261p, 263p and 525i
- Direct connection to TV with RGB-SCART cable shows complete grayish background with slightly less grayish text:

![SCART-RGB TV Grayish](https://github.com/somhi/PCXT_DeMiSTify/assets/1955895/650d2a38-78ac-4cbc-830f-c49ab1e526a9)

- What i observed is, that if i lower the threshold level for the Sync-on-Green signal on the RetroTink, the image stabilizes and works as expected:

![IMG_0475](https://github.com/somhi/PCXT_DeMiSTify/assets/1955895/825e21f7-60b5-457a-afdc-49f5349cbf5b)

**15khz, 15khz-1, 15-khz-2**
- Same behavior as with the officially release core.

**Comment by somhi** [comment]
does it make any difference switching debug options  on 15 khz branch ?

So can it be considered solved in your case with solution " lower the threshold level for the Sync-on-Green signal on the RetroTink," ?

**Comment by egrath** [comment]
Yes the debug options made a difference - when turning on the first one, i loose sync entirely ("no signal"). The second one made no difference except for a slight hop of the entire picture to the left.

As for the RetroTink: I would say it's more of a workaround for those who use the 15 Khz on one of those devices. For those using RGB-SCART directly on their TV it's not possible, as they can't tweak such parameters.

**Comment by somhi** [comment]
Ok thanks.

As for the direct connection, I have an RGB-SCART cable connected directly to an LCD monitor which accepts 15 kHz input, and I've seen no hopping problems. I just observed bending problems at top part of screen in some resolutions.

We should see other scaler devices like OSSC what they do, but according to these tests, unfortunately no clear answer have been found on what to change
